### PR TITLE
Document an overview of all our tests

### PR DESCRIPTION
**Pull request by cecilia-donnelly**
Add information about the test frameworks we have and how to write new tests.  This still needs to incorporate @jasonaowen's [comment](https://github.com/OpenTechStrategies/psm/pull/253#issuecomment-325821012) and more detail about Selenium from @BenGalewsky.

### CODE REVIEW

File: docs/TESTING.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+for more details on how to set up and run these tests.
+
+To add a new Selenium test, do the following:
+
+1. Write a new "feature" file describing the
+[`Given-When-Then`](http://thucydides.info/docs/serenity-staging/#_serenity_with_cucumber)
+flow of the new test.  These feature files are in this repository in
+`psm-app/integration-tests/src/test/resources/features`.
+
+2. Write the handlers to enter the information or click the buttons in
```
Comment (BenGalewsky) [inline docs/TESTING.md]: It's very handy to run the test suite after creating new scenarios in the feature file. Serenity helpfully writes out template code to the console for any statements that don't have matching step definitions. You can paste these template methods into your step definition class to make it easy to write the behavior.

Just remember to remove the `throws Throwable` annotation unhelpfully added to the template method definitions.

Comment (cecilia-donnelly) [reply docs/TESTING.md]: Oh that's really useful!  Thank you.  I'll try that and explain it in the TESTING doc.

**Comment by cecilia-donnelly** [comment]
I think this is ready to merge now.  If anyone has time to review, please do so!  If I don't hear anything I'll merge in a day or two -- this is docs-only, so if we need to change it later it will be fine.  As developer documentation, this change fits under the [post-commit-review proposal](https://groups.google.com/forum/#!topic/psm-dev/6ccg5K0UF2E).

File: docs/TESTING.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+When we add new functionality, we add unit tests along with it.  To
+specifically see the Spock tests, there are more specific Gradle tasks
+that can be run, like `./gradlew cms-business-process:check`.  You can
+also run `./gradlew tasks --all` to see all the tasks, and in particular
+there is a section labeled "Verification Tasks."  Note that Gradle will
+not run all the tests every time if it can see that neither the
+underlying code nor the tests have changed, so when you run `./gradlew
+clean test` you might not see test results.  However, those results are
+available.  Run `./gradlew cms-business-process:clean
+cms-business-process:test --info` to see where the reports are stored.
```
Comment (jasonaowen) [inline docs/TESTING.md]: I think I was a little unclear about this the other day. The `clean` in `./gradlew clean test` should cause all the tests to be rerun, and if the unit tests pass you won't see any output. As you correctly wrote, Gradle will not rerun the unit tests if there are no changes and you just do `./gradlew test`. In either case, it does not show you the location of the test reports unless you include the `--info` option.

So, I think the only thing that needs to be changed is:

```diff
 when you run
-`./gradlew clean test`
+`./gradlew test`
 you might not see test results.
```

Comment (cecilia-donnelly) [reply docs/TESTING.md]: Aha, thank you!  I'll fix that and then merge this. :)

**Pull request merged**